### PR TITLE
add a new blog 20190903

### PR DESCRIPTION
**Pull request by langyuemeng**
@breeswish @YiniXu9506 PTAL

### CODE REVIEW

File: tikv-source-code-reading-13.md [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+ 
+- 若外部只需要 1 个 KV 对（例如 limit = 1），此时就可以停止了，若外部还要继续获取更多 KV 对，则重新开始执行步骤 1。
+ 
+- 执行步骤 1：不是首次迭代，跳过。
+ 
+- 执行步骤 2：对比 Lock Cursor 与 Write Cursor，进入分支 2.4。
+ 
+- 执行分支 2.4：Write Cursor 指向 `foo`，Lock Cursor 指向 `box`，User Key 为 `box`。
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/9.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: 缺少一个空格(一共需要 2 个前导空格)

File: tikv-source-code-reading-13.md [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+ 
+<center>图 9 执行完毕后各个 Cursor 位置示意</center>
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: 缺少2个空格

File: tikv-source-code-reading-13.md [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+ 
+- 执行步骤 3：User Key = box 来自于 Lock，继续。
+ 
+- 执行步骤 3.1：检查 Lock Info。Lock 的 ts 为 0x11，`scan_ts` 为 0x05，忽略这个 Lock 不返回锁冲突错误。
+ 
+- 执行步骤 3.2：将 Lock Cursor 往后移动一个 Key。
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/10.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

File: tikv-source-code-reading-13.md [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+ 
+ <center>图 10 执行完毕后各个 Cursor 位置示意</center>
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

File: tikv-source-code-reading-13.md [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+
+ 
+- 执行步骤 4：User Key = box 不来自于 Write，跳过，回到步骤 2。
+ 
+- 执行步骤 2：对比 Lock Cursor 与 Write Cursor，进入分支 2.4。
+ 
+- 执行分支 2.4：Write Cursor 指向 `foo`，Lock Cursor 指向 `foo`，User Key 为 `foo`。
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/11.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

File: tikv-source-code-reading-13.md [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+ 
+ <center>图 11 执行完毕后各个 Cursor 位置示意</center>
+ 
+- 执行步骤 3：User Key = foo 来自于 Lock，继续。与之前类似，锁被忽略，且 Lock Cursor 往后移动。
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/12.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

File: tikv-source-code-reading-13.md [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+ 
+- 执行步骤 4：User Key = foo 同样来自于 Write，继续。
+ 
+- 执行步骤 4.1：Seek `{w_key}{!scan_ts}`，即 Seek `foo......\xFF\xFF..\xFA`。Write Cursor 仍然是当前位置。
+ 
+- 执行步骤 4.2：此时 Write Key 指向 foo 与 User Key 相同，因此依据 `PUT (start_ts=1)` 从 Default CF 中获取到 `value = foo_value`。
+ 
+- 执行步骤 4.3：移动 Write Cursor 跳过当前 `foo` 剩余所有版本，即 Seek `foo......\xFF\xFF..\xFF`：
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/13.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

File: tikv-source-code-reading-13.md [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+ 
+- 执行步骤 4.4：对外返回 Key Value 对 `(foo, foo_value)`。
+ 
+- 若外部选择继续扫，则继续回到步骤 1。
+ 
+- 执行步骤 1：不是首次迭代，跳过。
+ 
+- 执行步骤 2：对比 Lock Cursor 与 Write Cursor，进入分支 2.1。
+ 
+ ![执行完毕后各个 Cursor 位置示意](media/tikv-source-code-reading-13/14.png)
```
Comment (breeswish) [inline tikv-source-code-reading-13.md]: ditto

**Comment by YiniXu9506** [comment]
related to pr #291

**Pull request closed without merging**